tiingo/api.py

```diff
@@ -166,6 +166,47 @@ def _get_url(self, ticker, frequency):
             else:
                 return "iex/{}/prices".format(ticker)
 
+    def _request_pandas(self, ticker, metric_name, params):
+        """
+        Return data for ticker as a pandas.DataFrame if metric_name is not
+        specified or as a pandas.Series if metric_name is specified.
+
+        :param ticker (string): ticker to be requested
+        :param params (dict): a dictionary containing valid resampleFreq
+            and format strings per the Tiingo api
+        :param metric_name (string): Optional parameter specifying metric to be returned for each
+            ticker.  In the event of a single ticker, this is optional and if not specified
+            all of the available data will be returned.  In the event of a list of tickers,
+            this parameter is required.
+        """
+        url = self._get_url(ticker, params['resampleFreq'])
+        response = self._request('GET', url, params=params)
+        if params['format'] == 'csv':
+            if sys.version_info < (3, 0):  # python 2
+                from StringIO import StringIO
+            else:  # python 3
+                from io import StringIO
+
+            df = pd.read_csv(StringIO(response.content.decode('utf-8')))
+        else:
+            df = pd.DataFrame(response.json())
+
+        df.set_index('date', inplace=True)
+
+        if metric_name is not None:
+            prices = df[metric_name]
+        else:
+            prices = df
+
+        prices.index = pd.to_datetime(prices.index)
+
+        # Localize to UTC to ensure equivalence between data returned in json format and
+        # csv format. Data requested in csv format does not include a timezone.
```

Comment:
> Data requested in csv format does not include a timezone

Nice catch, I was curious about whether there was still any benefit to using `json` format if CSV has so much better performance. 

[non-blocking] @tiingo are there any other major differences between the data returned by each format that users should know about besides query performance and the presence of timezones?

Reply:
I should specify more clearly that daily data returned by Tiingo does not include a timezone. For intraday data, a timezone is included with the csv response.

This timezone issue exposes a larger issue which I think we should discuss. Tiingo daily data and IEX intraday data are both being returned by the `get_dataframe` method but I think they would be better off with their own methods. This would follow the Tiingo API more closely and users would be more aware of the source of the underlying data. I would suggest replacing `get_dataframe` with `get_daily_dataframe` and `get_iex_dataframe`.